### PR TITLE
[core] feat(Callout): visual modernisation

### PR DESCRIPTION
**Pull request by CPerinet**
Proposition for modernisation of callouts. Reduce unnecessary visual drag while retaining prominence.
- More consistent and slightly increased spacings
- Reduce header size from `h4` to `h5`
- Reduce icon size from `20px` to `16px`
- Use intent colour as default text colour.

| Before | After |
|---|---|
|Screenshot 2023-01-10 at 12 08 42|Screenshot 2023-01-10 at 12 08 23|
|Screenshot 2023-01-10 at 12 11 02|Screenshot 2023-01-10 at 12 10 51|

### Accessibility checks

Given background colours uses opacity, I chose to evaluate accessibility with the callout on a `$white` / `$light-gray5` background for light theme and `$dark-gray1` / `$dark-gray2` for dark theme. 

| Theme | Intent | WCAG Normal Text |
|---|---|---|
| Light | Default | AA |
| Light | Primary | AA |
| Light | Success | AA |
| Light | Warning | AA |
| Light | Danger | AA |
| Dark | Default | AA |
| Dark | Primary | AA |
| Dark | Success | AA |
| Dark | Warning | AA |
| Dark | Danger | AA |

### CODE REVIEW

File: packages/core/src/components/callout/_callout.scss [hunk 1]
```diff
@@ -8,7 +8,7 @@ Callout
 
 Markup:
 <div class="#{$ns}-callout {{.modifier}}">
-  <h4 class="#{$ns}-heading">Callout Heading</h4>
+  <h5 class="#{$ns}-heading">Callout Heading</h5>
```
Comment (CPerinet) [inline packages/core/src/components/callout/_callout.scss:11]: In line with the [dialog update](https://github.com/palantir/blueprint/pull/5753), reducing the heading level to fit better in dense UIs.

File: packages/core/src/components/callout/_callout.scss [hunk 2]
```diff
@@ -25,20 +25,20 @@ Styleguide callout
   @include running-typography();
   background-color: rgba($gray3, 0.15);
   border-radius: $pt-border-radius;
-  padding: $pt-grid-size ($pt-grid-size * 1.2) ($pt-grid-size * 0.9);
+  padding: $pt-grid-size * 1.5;
```
Comment (CPerinet) [inline packages/core/src/components/callout/_callout.scss]: - Make the spacing consistent all around.
- Increase inner spacing.

File: packages/core/src/components/callout/_callout.scss [hunk 3]
```diff
@@ -25,20 +25,20 @@ Styleguide callout
...
   position: relative;
   width: 100%;
 
   // CSS API support
   &[class*="#{$ns}-icon-"] {
-    padding-left: ($pt-grid-size * 2) + $pt-icon-size-large;
+    padding-left: ($pt-grid-size * 2.2) + $pt-icon-size-standard;
```
Comment (CPerinet) [inline packages/core/src/components/callout/_callout.scss:34]: 2.2 is 1.5 (consistent side padding) and 0.7 for spacing between icon and text (consistent with buttons).

File: packages/core/src/components/callout/_callout.scss [hunk 4]
```diff
@@ -93,6 +97,7 @@ Styleguide callout
 
       .#{$ns}-dark & {
         background-color: rgba($color, 0.2);
+        color: map-get($pt-dark-intent-text-colors, $intent);
```
Comment (CPerinet) [inline packages/core/src/components/callout/_callout.scss:100]: Make default text colour intent. Adding accessibility checks

**Review by adidahiya — APPROVED** [review]

**Comment by adidahiya** [comment]
Color changes LGTM. I also tested contrast without the docs example container background (by default, it increases the apparent contrast), because I think `$pt-app-background-color` and `$pt-dark-app-background-color` are more representative of real UI scenarios:

image

**Comment by adidahiya** [comment]
Docs preview: https://output.circle-artifacts.com/output/job/9906788b-45b8-4d82-b4c9-2a8148f05dd3/artifacts/0/packages/docs-app/dist/index.html#core/components/callout

**Pull request merged**